torch/csrc/autograd/custom_function.cpp

```diff
@@ -260,6 +261,7 @@ optional_variable_list _process_backward_mode_ad(
       // Run in no_grad mode to mimic the behavior of the forward.
       {
         AutoGradMode grad_mode(false);
+        at::AutoFwGradMode fw_grad_mode(false);
```

Comment:
It turns out that this particular output already has the forward grad we desired!

So... an alternate solution is actually to not have this guard, but if we detect `is_input && !mark_dirty`, then we simply continue instead of setting fwgrad.

Reply:
Yes that sounds like the right thing to do!